tools/mage/dev.go

```diff
@@ -82,6 +87,73 @@ func execDockerCompose(args ...string) error {
 	return err
 }
 
+func execDockerComposeWithOutput(filepath string, args ...string) error {
+	output, err := sh.Output("docker-compose", dockerComposeFlags(args...)...)
+	if err != nil {
+		return err
+	}
+	message := []byte(output)
+	err = ioutil.WriteFile(filepath, message, 0644)
+	return err
+}
+
+// SQLStart starts the SQL database of the development environment.
+func (Dev) SQLStart() error {
+	if mg.Verbose() {
+		fmt.Printf("Starting SQL databases\n")
+	}
+	if err := execDockerCompose(append([]string{"up", "-d"}, sqlDatabase)...); err != nil {
+		return err
+	}
+	return execDockerCompose("ps")
+}
+
+// SQLStop stops the SQL database of the development environment.
+func (Dev) SQLStop() error {
+	if mg.Verbose() {
+		fmt.Printf("Stopping SQL databases\n")
+	}
+	return execDockerCompose(append([]string{"stop"}, sqlDatabase)...)
+}
+
+// SQLMakeSnapshot stores the current cockroach data folder for later restores.
+func (Dev) SQLMakeSnapshot() error {
+	if mg.Verbose() {
+		fmt.Printf("Making DB snapshot")
+	}
+	os.RemoveAll(devDataDir + "/cockroach-snap")
+	return sh.RunV("cp", "-R", devDataDir+"/cockroach", devDataDir+"/cockroach-snap")
```

Comment:
IDEM, also prefer using stdlib instead of shelling out to `cp` - different environments may have different implementations and even no `cp` at all (e.g. Windows).